### DISCUSSION TITLE
No such file phpqrcode/qrmask.php on line 155

### DISCUSSION THREAD

**Discussion by saxrub** [chillerlan/php-qrcode#254] [General]
HI all

When i build a QRCode, il get this erreor in my PHP error log.

PHP Warning:  file_put_contents(../tmp/mask_6/mask_41_6.dat): failed to open stream: No such file or directory in /phpqrcode/qrmask.php on line 155

In the site folders, the phpqrcode folder is directly under www.

At the same level there are folders named tmp, temp and tmp_qrcode

In the phpqrcode folder there are folders named temp and tmp.

In qrconfig.php we find
define('QR_CACHE_DIR', '..'.DIRECTORY_SEPARATOR.'tmp'.DIRECTORY_SEPARATOR);
define('QR_LOG_DIR', '..'.DIRECTORY_SEPARATOR); // default error logs dir
    
Note that the QRCode is displayed correctly.

What can I do to remove this warning?

Pascal

**Comment by codemasher** [comment]
Are you sure you're reporting to the right library? I know the maintainer of phpqrcode has recommended to use this library here, but I'm not much of a help when it comes to issues with their library.

https://github.com/t0k4rt/phpqrcode/blob/master/qrmask.php
